packages/model-viewer/src/features/environment.ts

```diff
@@ -126,6 +126,27 @@ export const EnvironmentMixin = <T extends Constructor<ModelViewerElementBase>>(
         this[$cancelEnvironmentUpdate] = null;
       }
 
+      // Update background color before to get texture utils because renderer singleton depends on background transparency
+      if (skyboxImage == null)
+      {
+        if (backgroundColor === 'transparent') {
+          if (this[$scene].background != null) {
+            this.disconnectedCallback();
```

Comment:
These (`connectedCallback`/`disconnectedCallback`) are custom element lifecycle methods, and are meant to be invoked by the browser based on the semantics defined in the custom element spec. If we are calling them manually, then it's a sign that we may need to factor out part of their implementation to discrete methods.

Why are we invoking these methods manually?

Reply:
These callbacks are called to unregister the scene from the renderer and to unsubscribe from the contextlost event. I change the code to call only what is necessary.